src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/span
+    /// <summary>
+    /// Transaction span.
+    /// </summary>
+    public class Span : ISpan, IJsonSerializable
+    {
+        /// <inheritdoc />
+        public SentryId SpanId { get; }
+
+        /// <inheritdoc />
+        public SentryId? ParentSpanId { get; }
+
+        /// <inheritdoc />
+        public SentryId TraceId { get; private set; }
+
+        /// <inheritdoc />
+        public DateTimeOffset StartTimestamp { get; private set; } = DateTimeOffset.UtcNow;
+
+        /// <inheritdoc />
+        public DateTimeOffset? EndTimestamp { get; private set; }
+
+        /// <inheritdoc />
+        public string Operation { get; }
+
+        /// <inheritdoc />
+        public string? Description { get; set; }
+
+        /// <inheritdoc />
+        public SpanStatus? Status { get; private set; }
+
+        /// <inheritdoc />
+        public bool IsSampled { get; set; }
+
+        private ConcurrentDictionary<string, string>? _tags;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new ConcurrentDictionary<string, string>();
+
+        private ConcurrentDictionary<string, object?>? _data;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, object?> Extra => _data ??= new ConcurrentDictionary<string, object?>();
```

Comment:
Does a span really have to be thread-safe? Maybe only the transaction needs to be which would make our life's easier.

Otherwise this assignment again isn't thread-safe